src/browser/modules/D3Visualization/components/Graph.tsx

```diff
@@ -123,6 +133,13 @@ export class GraphComponent extends React.Component<GraphProps> {
     }
   }
 
+  handleZoomEvent = (limitsReached: ZoomLimitsReached) => {
+    this.setState({
```

Comment:
Since this is called every time the zoom changes, it can trigger loads of re-renders, especially when zooming with the scroll wheel. It might be worth setting the state only if it changed.

Reply:
Nice catch! I'll fix!